### ISSUE TITLE
Latest aws-sdk-core release (3.165.0) breaks aws-sdk-s3 @ 1.114.0

### ISSUE THREAD

**Issue by skevy** [aws/aws-sdk-ruby#2774]
### Describe the bug

After the latest release (3.165.0) of aws-sdk-core, I started getting this error:

```
./aws-sdk-s3/plugins/s3_signer.rb:59:in `call': uninitialized constant Aws::Plugins::SignatureV4 (NameError)
```

### Expected Behavior

This error does not occur.

### Current Behavior

A fatal error when using the S3 client.

### Reproduction Steps

Simply calling `Aws::S3::Bucket.new(` will cause this.

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

aws-sdk-core

### Environment details (Version of Ruby, OS environment)

Ruby 2.6, Linux

**Comment by alextwoods** [comment]
Sorry you're running into this - I'm looking into it.can you provide more information on all of the gem versions in your environment and some example code?

**Comment by skevy** [comment]
```
Successfully installed aws-sigv4-1.5.2
Successfully installed aws-sdk-core-3.165.0
Successfully installed aws-sdk-kms-1.59.0
Successfully installed aws-sdk-s3-1.114.0
```

These are the aws versions I have installed. Any invocation of the AWS S3 ruby client causes this error.

**Comment by skevy** [comment]
I'm pretty sure this is caused due to an incompatibility between aws-sdk-s3-1.114.0 and the latest version of aws-sdk-core

**Comment by alextwoods** [comment]
I'm still investigating the root cause and a fix, but in the mean time you can either: upgrade to latest aws-sdk-s3 or downgrade your core version.

**Comment by skevy** [comment]
Yah we've worked around by upgrading everything (and locking), but I reported because I presume it wasn't the intention to break backwards compat here. So I wanted to make sure it was brought to your attention.

**Comment by alextwoods** [comment]
Yeah - we definintly did not intent to break compatibiltity.  Weirdly it works with the older s3 gem AND the older kms gem together (which is how we had verified this).  Using the newer kms (1.59.0 as you had) + an older s3 gem breaks....

**Comment by alextwoods** [comment]
Okay - we've released aws-sdk-core 1.165.1 which should address this issue.

Thanks again for reporting this and sorry you were impacted.

**Comment by skevy** [comment]
Thanks folks.
